rmw_zenoh_cpp/src/detail/zenoh_utils.cpp

```diff
@@ -42,13 +42,18 @@ ZenohSession::~ZenohSession()
 
 ///=============================================================================
 zenoh::Bytes create_map_and_set_sequence_num(
-  int64_t sequence_number, std::array<uint8_t, RMW_GID_STORAGE_SIZE> gid)
+  int64_t sequence_number,
+  std::array<uint8_t, RMW_GID_STORAGE_SIZE> gid,
+  int64_t * source_timestamp)
```

Comment:
I see why you've done this, but I'm honestly not a fan of this change to the API.  It changes all of the arguments from being input-only, to one of them being an out parameter.

I'm going to propose an alternate solution here as a PR targeting this one.

Reply:
See https://github.com/christophebedard/rmw_zenoh/pull/1

Reply:
yeah that's great, thank you!